### PR TITLE
TST: Avoid bare pytest.raises in test_series.py

### PR DESCRIPTION
**Pull request by Vlek**
* [x]  ref #30999
 
* [x]  tests added / passed
 
* [x]  passes `black pandas`

* [x]  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Please note that this file had an error message that was pretty ugly, so I changed
it to only include the class name as I have done for other tests.

### CODE REVIEW

File: pandas/tests/dtypes/test_dtypes.py [hunk 1]
```diff
@@ -414,25 +414,22 @@ def test_construction_from_string(self, dtype):
         assert is_dtype_equal(dtype, result)
         result = PeriodDtype.construct_from_string("period[D]")
         assert is_dtype_equal(dtype, result)
+        msg = "Cannot construct a 'PeriodDtype' from "
```
Comment (jreback) [inline pandas/tests/dtypes/test_dtypes.py]: can you parameterize these (like they were); also can add whats on line 430 in the same test

File: pandas/core/generic.py [hunk 2]
```diff
@@ -354,7 +354,7 @@ def _get_axis_number(cls, axis):
                 return cls._AXIS_NUMBERS[axis]
             except KeyError:
                 pass
-        raise ValueError(f"No axis named {axis} for object type {cls}")
+        raise ValueError(f"No axis named {axis} for object type {cls.__name__}")
```
Comment (jreback) [inline pandas/core/generic.py]: can you put ``<{cls.__name__}>``

File: pandas/core/generic.py [hunk 3]
```diff
@@ -367,7 +367,7 @@ def _get_axis_name(cls, axis):
                 return cls._AXIS_NAMES[axis]
             except KeyError:
                 pass
-        raise ValueError(f"No axis named {axis} for object type {cls}")
+        raise ValueError(f"No axis named {axis} for object type {cls.__name__}")
```
Comment (jreback) [inline pandas/core/generic.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
this is a very tiny api change, but ok.

File: pandas/core/generic.py (hunk 2 above)
Comment (Vlek) [reply pandas/core/generic.py]: I'm following the format that I was given by @jorisvandenbossche in this pull request: https://github.com/pandas-dev/pandas/pull/32816

File: pandas/tests/dtypes/test_dtypes.py (hunk 1 above)
Comment (Vlek) [reply pandas/tests/dtypes/test_dtypes.py]: That was unfortunately a change made in a different pull request and my ide keeps bringing those commits with each new branch I make. I had added the parameterized checks in the beginning, so I have pasted that code back in. Unfortunately for that last check, it has a different type of error message that wouldn't be encapsulated with the error message and it wouldn't fit the parameterized pytest check datatype as everything else is of type string.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**